nixos/tests/evcc.nix

```diff
@@ -0,0 +1,68 @@
+import ./make-test-python.nix ({ pkgs, lib, ...} :
+
+{
+  name = "evcc";
+  meta.maintainers = with lib.maintainers; [ hexa ];
+
+  nodes = {
+    machine = { config, ... }: {
+      services.evcc = {
+        enable = true;
+        settings = {
+          uri = "localhost:7070";
+          log = "error";
+
+          site = {
+            title = "NixOS Test";
+            meters = {
+              grid = "grid";
+              pv = "pv";
+            };
+          };
+
+          meters = [ {
+            type = "custom";
+            name = "grid";
+            power = {
+              source = "script";
+              cmd = "/bin/sh -c 'echo -4500'";
+            };
+          } {
+            type = "custom";
+            name = "pv";
+            power = {
+              source = "script";
+              cmd = "/bin/sh -c 'echo 7500'";
+            };
+          } ];
+
+          chargers = [ {
+            name = "dummy-charger";
+            type = "keba";
+            uri = "127.0.0.1";
+          } ];
+
+          loadpoints = [ {
+            title = "Dummy";
+            charger = "dummy-charger";
+          } ];
+        };
+      };
+    };
+  };
+
+  testScript = ''
+    start_all()
+
+    machine.wait_for_unit("evcc.service")
+    machine.wait_for_open_port(7070)
+
+    machine.succeed(
+        "curl --fail http://localhost:7070"
+    )
+
+    with subtest("Check systemd hardening"):
+        output = machine.execute("systemd-analyze security evcc.service | grep -v '✓'")[1]
```

Comment:
Ugh, this might work today but this might break whenever we upgrade systemd and they change their output format. Is there perhaps a machine readable format that you should consume instead? Also not that the severity levels might change when upgrading systemd.

I've done similar tests myself in the past but usually just during packaging to ensure I do not accidentially change something on there.

Reply:
Pretty sure that is the fallback format already, given that the normal format has emojis.

Basically the grep is there to cut down on the noise. And the test itself is there, so I can quickly see on updates if there is something I could easily improve.

Reply:
But still if systemd-analyze adds new checks it should not break the service. For the same reason we don't enable flake8 linting as upgrading it, can break perfectly fine packages.

Reply:
Why is machine.execute and issue then? It is allowed to fail if I don't misunderstand it.

Reply:
Never mind! I got that wrong.